### PR TITLE
Vagrant pdf build

### PR DESCRIPTION
**Pull request by saxocellphone**
This builds the pdf-annotate.js code and copies it to the site/js/pdf folder.

### CODE REVIEW

File: .setup/install_system.sh [hunk 1]
```diff
@@ -378,6 +378,13 @@ EOF
     chmod 0640 /etc/apache2/suexec/www-data
 
 
+	#############
+	#NPM install
+	#############
+
+	curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+	sudo apt-get install -y nodejs
```
Comment (MasterOdin) [inline .setup/install_system.sh]: you don't need sudo, the script has to be run as root.

Though, please convert the tabs to spaces.

**Comment by saxocellphone** [comment]
It should be working now but someone else should run a vagrant up just to make sure

**Comment by saxocellphone** [comment]
Now that I took a look at how other people do version releases, I think there might be a better approach. Instead of building it when installing submitty, we should just copy the file in the /dist folder in the other repo. When we want to update the code, we can just make a new release in the other repo and update the version number.

**Comment by MasterOdin** [comment]
The reason for the `*.js.map` file is to map the compiled source to where you would look in the uncompiled version for debugging purposes. As such, there's no real reason not to just always use the `*.min.js` version over unminified, so I removed that logic.

I also haven't really tested the PDF functionality, so it'd be worth confirming things still work after my changes.

**Comment by bmcutler** [comment]
@saxocellphone Can you please resolve conflicts?

**Pull request merged**